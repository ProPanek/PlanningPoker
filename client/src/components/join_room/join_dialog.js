import React, {Component} from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import TextField from '@material-ui/core/TextField'
import DialogActions from '@material-ui/core/DialogActions'
import Button from '@material-ui/core/Button'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import {decorate, observable} from "mobx";


class JoinDialog extends Component {
  handleChange = (e) => {
    this.input = e.target.value
  }
  joinRoom = (name) => {
    const {match: {params}} = this.props
    this.props.store.joinRoom(params.id, params.password, name)
    this.props.store.openJoinDialog = false
    this.props.store.connected = true
  }

  cancelJoinRoom = () => {
    this.props.history.push(`/`)
  }

  render() {

    return (
      <Dialog
        open={this.props.store.openJoinDialog}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Join Room</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please write your username below if you want to join to the room.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="User Name"
            type="text"
            fullWidth
            value={this.input}
            onChange={this.handleChange}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.cancelJoinRoom} color="primary">
            Cancel Join
          </Button>
          <Button onClick={() => {
            this.joinRoom(this.input)
          }} color="primary" variant="contained">
            Join Room
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

decorate(JoinDialog, {
  input: observable
});
export default inject("store")(withRouter(observer(JoinDialog)));

// export default JoinDialog