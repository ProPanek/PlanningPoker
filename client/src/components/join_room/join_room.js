import React, { Component } from "react";
import { Grid, Card } from "@material-ui/core";
import styled from "styled-components";
import { inject, observer } from "mobx-react";

const StyledGrid = styled(Grid)`
  height: calc(100vh - 64px);
`;

const StyledCard = styled(Card)`
  height: calc(100vh - 64px);
`;

class JoinRoom extends Component {
  render() {
    return (
      <StyledGrid item xs={6}>
        <StyledCard>
          <select>
            {/*{console.log(this.props.store.rooms)}*/}
            {this.props.store.rooms.map((data, index) => {
              return <option key={index}>{data.roomName}</option>;
            })}
          </select>
        </StyledCard>
      </StyledGrid>
    );
  }
}

// export default JoinRoom;
export default inject("store")(observer(JoinRoom));
