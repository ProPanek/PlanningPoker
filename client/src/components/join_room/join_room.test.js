import React from "react";
import {JoinRoom} from "./join_room";
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe("JoinRoom", () => {

  it("renders without crashing", () => {
    const store = {
      jira:{
        jiraBoardsFetching: true,
        jiraBoards: { values: []}
      },
      room:{rooms:[]}
    }
    const component = shallow(<JoinRoom store={store}/>)
    expect(component).toMatchSnapshot()
  });
});