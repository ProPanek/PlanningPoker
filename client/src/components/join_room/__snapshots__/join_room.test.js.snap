// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinRoom renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JoinRoom
    store={
      Object {
        "jira": Object {
          "jiraBoards": Object {
            "values": Array [],
          },
          "jiraBoardsFetching": true,
        },
        "room": Object {
          "rooms": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          <select
            onClick={[Function]}
            size="10"
          />
          <TextField
            id="join-user-name"
            label="User Name"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
            variant="standard"
          />
          <TextField
            id="join-room-password"
            label="Room Password"
            onChange={[Function]}
            required={false}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <ForwardRef />
          <WithStyles(Button)
            onClick={[Function]}
          >
            Join room
          </WithStyles(Button)>
          <WithStyles(Button)
            onClick={[Function]}
          >
            Delete room
          </WithStyles(Button)>
        </ForwardRef>
      </ForwardRef>,
      "item": true,
      "xs": 6,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <select
            onClick={[Function]}
            size="10"
          />
          <TextField
            id="join-user-name"
            label="User Name"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
            variant="standard"
          />
          <TextField
            id="join-room-password"
            label="Room Password"
            onChange={[Function]}
            required={false}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <ForwardRef />
          <WithStyles(Button)
            onClick={[Function]}
          >
            Join room
          </WithStyles(Button)>
          <WithStyles(Button)
            onClick={[Function]}
          >
            Delete room
          </WithStyles(Button)>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <select
              onClick={[Function]}
              size="10"
            />,
            <TextField
              id="join-user-name"
              label="User Name"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              value=""
              variant="standard"
            />,
            <TextField
              id="join-room-password"
              label="Room Password"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              value=""
              variant="standard"
            />,
            <ForwardRef />,
            false,
            <WithStyles(Button)
              onClick={[Function]}
            >
              Join room
            </WithStyles(Button)>,
            <WithStyles(Button)
              onClick={[Function]}
            >
              Delete room
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": false,
              "onClick": [Function],
              "size": "10",
            },
            "ref": null,
            "rendered": false,
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "join-user-name",
              "label": "User Name",
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": "",
              "variant": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "join-room-password",
              "label": "Room Password",
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "password",
              "value": "",
              "variant": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": [Function],
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  &&{
  margin: 0 auto;
  }
",
                ],
              },
              "displayName": "Styled(WithStyles(CircularProgress))",
              "options": Object {
                "flip": false,
                "name": "MuiCircularProgress",
              },
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Join room",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Join room",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete room",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete room",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: 0 auto;
  position: relative;
  top: calc(50vh - 144px);
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Naked": [Function],
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  height: calc(100vh - 48px);
",
          ],
        },
        "displayName": "Styled(WithStyles(Card))",
        "options": Object {
          "name": "MuiCard",
        },
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Naked": [Function],
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  height: calc(100vh - 48px);
",
        ],
      },
      "displayName": "Styled(WithStyles(Grid))",
      "options": Object {
        "name": "MuiGrid",
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <select
              onClick={[Function]}
              size="10"
            />
            <TextField
              id="join-user-name"
              label="User Name"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              value=""
              variant="standard"
            />
            <TextField
              id="join-room-password"
              label="Room Password"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              value=""
              variant="standard"
            />
            <ForwardRef />
            <WithStyles(Button)
              onClick={[Function]}
            >
              Join room
            </WithStyles(Button)>
            <WithStyles(Button)
              onClick={[Function]}
            >
              Delete room
            </WithStyles(Button)>
          </ForwardRef>
        </ForwardRef>,
        "item": true,
        "xs": 6,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <select
              onClick={[Function]}
              size="10"
            />
            <TextField
              id="join-user-name"
              label="User Name"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              value=""
              variant="standard"
            />
            <TextField
              id="join-room-password"
              label="Room Password"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              value=""
              variant="standard"
            />
            <ForwardRef />
            <WithStyles(Button)
              onClick={[Function]}
            >
              Join room
            </WithStyles(Button)>
            <WithStyles(Button)
              onClick={[Function]}
            >
              Delete room
            </WithStyles(Button)>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <select
                onClick={[Function]}
                size="10"
              />,
              <TextField
                id="join-user-name"
                label="User Name"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="standard"
              />,
              <TextField
                id="join-room-password"
                label="Room Password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
                variant="standard"
              />,
              <ForwardRef />,
              false,
              <WithStyles(Button)
                onClick={[Function]}
              >
                Join room
              </WithStyles(Button)>,
              <WithStyles(Button)
                onClick={[Function]}
              >
                Delete room
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": false,
                "onClick": [Function],
                "size": "10",
              },
              "ref": null,
              "rendered": false,
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "join-user-name",
                "label": "User Name",
                "margin": "normal",
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "join-room-password",
                "label": "Room Password",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Naked": [Function],
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
  &&{
  margin: 0 auto;
  }
",
                  ],
                },
                "displayName": "Styled(WithStyles(CircularProgress))",
                "options": Object {
                  "flip": false,
                  "name": "MuiCircularProgress",
                },
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Join room",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Join room",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete room",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete room",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: 0 auto;
  position: relative;
  top: calc(50vh - 144px);
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Naked": [Function],
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  height: calc(100vh - 48px);
",
            ],
          },
          "displayName": "Styled(WithStyles(Card))",
          "options": Object {
            "name": "MuiCard",
          },
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Naked": [Function],
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  height: calc(100vh - 48px);
",
          ],
        },
        "displayName": "Styled(WithStyles(Grid))",
        "options": Object {
          "name": "MuiGrid",
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
